InventoryInStorePickupShipping/Model/ResourceModel/GetPickupLocationIntersectionForSkues.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupShipping\Model\ResourceModel;
+
+use Magento\Framework\DB\Select;
+use Magento\Framework\DB\Sql\ExpressionFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Inventory\Model\ResourceModel\Source;
+use Magento\Inventory\Model\ResourceModel\SourceItem;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryInStorePickupApi\Api\Data\PickupLocationInterface;
+
+/**
+ * Provides list of Source Codes which have all requested products assigned.
+ */
+class GetPickupLocationIntersectionForSkues
```

Comment:
Did you mean `GetPickupLocationIntersectionForSkus`?

Reply:
I am not sure if this class is related to any intersections for real.
It searches active pickup locations by SKUs, isn't it?

Reply:
It provides list of pickup locations which have intersection of assignment of provided in request SKUs.

Reply:
What about `GetAsignedPickupLocationsBySkus` or something like that? The class name should not sound like it contains something from the quantum physics inside

Reply:
I think name `GetAsignedPickupLocationsBySkus` does not represents what it actually do.

Reply:
Please use `GetPickupLocationIntersectionForSkus`

Reply:
Done!